lib/uwb/include/uwb/UwbMacAddress.hxx

```diff
@@ -348,9 +349,19 @@ public:
     static UwbMacAddress
     Random(UwbMacAddressType type);
 
+    /**
+     * @brief Creates a set of UwbMacAddress from a comma-delimited string of multiple mac addresses.
+     *
+     * @param addressesString The comma-delimited string of mac addresses.
+     * @param addressType The type of each mac address.
+     * @return std::optional<std::unordered_set<UwbMacAddress>> The constructed optional set of UwbMacAddress.
+     */
+    static std::optional<std::unordered_set<UwbMacAddress>>
+    MacAddressesFromString(const std::string& addressesString, UwbMacAddressType addressType);
```

Comment:
Thanks for the insight. I think it makes most sense to put this in `tools/nocli`. We really only need to parse a string of mac addresses into multiple addresses for this very specific purpose, in which we expect each one to be unique. So rather than keeping it in `UwbMacAddress` and making it a templatized version for multiple container types, I'll just move it into the `nocli` code. I also don't think we'll need the string parsing anywhere else, so I won't put it in `lib/uwb/protocols/fira`, but it can be moved there in the future if there ever becomes a need.

Reply:
Sounds perfect.